pkg/resource/logging-credentials/logging_operator_secrets.go

```diff
@@ -0,0 +1,75 @@
+package loggingcredentials
+
+import (
+	"math/rand"
+	"strings"
+	"time"
+
+	loggedcluster "github.com/giantswarm/logging-operator/pkg/logged-cluster"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// LoggingCredentialsSecretMeta returns metadata for the logging-operator credentials secret.
+func LoggingCredentialsSecretMeta(lc loggedcluster.Interface) metav1.ObjectMeta {
+	return metav1.ObjectMeta{
+		Name:      "logging-credentials",
+		Namespace: "monitoring",
+	}
+}
+
+// Generate a random 20-characters password
+func genPassword() string {
+	const length = 20
+
+	rand.Seed(time.Now().UnixNano())
+	chars := []rune("ABCDEFGHIJKLMNOPQRSTUVWXYZ" +
+		"abcdefghijklmnopqrstuvwxyz" +
+		"0123456789" +
+		"~=+%^*/()[]{}/!@#$?|")
+
+	var b strings.Builder
+	for i := 0; i < length; i++ {
+		b.WriteRune(chars[rand.Intn(len(chars))])
+
+	}
+	str := b.String() // E.g. "ExcbsVQs"
+	return str
+}
+
+// GenerateObservabilityBundleConfigMap returns a configmap for
+// the observabilitybundle application to enable promtail.
+func GenerateLoggingCredentialsBasicSecret(lc loggedcluster.Interface) (v1.Secret, error) {
+
+	secret := v1.Secret{
+		ObjectMeta: LoggingCredentialsSecretMeta(lc),
```

Comment:
It would be cool to have some label with information about what component is managing the `Secret`, so that we can know if a secret was manually created or it was created by some automation.

Reply:
Added with commit 476d19649ac56e906abe50833d8f4585e8aa6e2e